tokio/src/sync/batch_semaphore.rs

```diff
@@ -41,15 +41,21 @@ struct Waitlist {
     closed: bool,
 }
 
-/// Error returned by `Semaphore::try_acquire`.
+/// Error returned from the `Semaphore::try_acquire` function.
 #[derive(Debug)]
-pub(crate) enum TryAcquireError {
+pub enum TryAcquireError {
+    /// The semaphore has been closed and cannot issue new permits
     Closed,
+
+    /// The has no available permits.
     NoPermits,
 }
 /// Error returned by `Semaphore::acquire`.
+///
+/// An `acquire` operation can only fail if the semaphore has been
+/// closed.
```

Comment:
might be worth linking to the close method here too?
```suggestion
/// [closed].
///
/// [closed]: crate::sync::Semaphore::close
```